test/sql-luatest/constraint_test.lua

```diff
@@ -37,3 +37,93 @@ g.test_constraints_1 = function()
+        ck1.eight = func_id
+        t.assert_equals(s.foreign_key, fk0)
+        t.assert_equals(s.constraint, ck0)
+        t.assert_equals(s:format()[1].constraint, ck1)
+        t.assert_equals(s:format()[2].foreign_key, fk1)
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "one";]])
+        t.assert_equals(s.foreign_key, {two = {field = {b = 'b'}}})
+        t.assert_equals(s.constraint, ck0)
+        t.assert_equals(s:format()[1].constraint, ck1)
+        t.assert_equals(s:format()[2].foreign_key, fk1)
+
+        local _, err = box.execute([[ALTER TABLE "a" DROP CONSTRAINT "one";]])
+        local res = [[Constraint 'one' does not exist in space 'a']]
+        t.assert_equals(err.message, res)
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "four";]])
+        t.assert_equals(s.foreign_key, {two = {field = {b = 'b'}}})
+        t.assert_equals(s.constraint, {three = func_id})
+        t.assert_equals(s:format()[1].constraint, ck1)
+        t.assert_equals(s:format()[2].foreign_key, fk1)
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "seven";]])
+        t.assert_equals(s.foreign_key, {two = {field = {b = 'b'}}})
+        t.assert_equals(s.constraint, {three = func_id})
+        t.assert_equals(s:format()[1].constraint, {eight = func_id})
+        t.assert_equals(s:format()[2].foreign_key, fk1)
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "two";]])
+        t.assert_equals(s.foreign_key, nil)
+        t.assert_equals(s.constraint, {three = func_id})
+        t.assert_equals(s:format()[1].constraint, {eight = func_id})
+        t.assert_equals(s:format()[2].foreign_key, fk1)
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "five";]])
+        t.assert_equals(s.foreign_key, nil)
+        t.assert_equals(s.constraint, {three = func_id})
+        t.assert_equals(s:format()[1].constraint, {eight = func_id})
+        t.assert_equals(s:format()[2].foreign_key, {six = {field = 'b'}})
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "eight";]])
+        t.assert_equals(s.foreign_key, nil)
+        t.assert_equals(s.constraint, {three = func_id})
+        t.assert_equals(s:format()[1].constraint, nil)
+        t.assert_equals(s:format()[2].foreign_key, {six = {field = 'b'}})
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "three";]])
+        t.assert_equals(s.foreign_key, nil)
+        t.assert_equals(s.constraint, nil)
+        t.assert_equals(s:format()[1].constraint, nil)
+        t.assert_equals(s:format()[2].foreign_key, {six = {field = 'b'}})
+
+        box.execute([[ALTER TABLE "a" DROP CONSTRAINT "six";]])
+        t.assert_equals(s.foreign_key, nil)
+        t.assert_equals(s.constraint, nil)
+        t.assert_equals(s:format()[1].constraint, nil)
+        t.assert_equals(s:format()[2].foreign_key, nil)
+
+        local _, err = box.execute([[ALTER TABLE "a" DROP CONSTRAINT "eight";]])
+        local res = [[Constraint 'eight' does not exist in space 'a']]
```

Comment:
You redefine `_, err, res` here. They are already declared above. Might be better to drop `local` keyword here.

Reply:
Thanks, fixed.